pkg/tide/github.go

```diff
@@ -583,3 +657,65 @@ func (mc *mergeChecker) prMergeMethod(c config.Tide, crc *CodeReviewCommon) *typ
 	}
 	return &method
 }
+
+type prowJobsByContext struct {
+	successfulBatchJob   map[string]prowapi.ProwJob
+	pendingPresubmitJobs map[string]bool
+}
+
+func getProwJobsForPRs(prs []CodeReviewCommon, pjs []prowapi.ProwJob) map[int]prowJobsByContext {
+	prowJobsForPRs := make(map[int]prowJobsByContext)
+	prNums := make(map[int]CodeReviewCommon)
+	for _, pr := range prs {
+		prNums[pr.Number] = pr
+	}
+
+	for _, pj := range pjs {
+		jobState := toSimpleState(pj.Status.State)
+
+		switch pj.Spec.Type {
+		case prowapi.BatchJob:
+			if jobState != successState {
+				continue
+			}
+			validPulls := true
+			for _, pull := range pj.Spec.Refs.Pulls {
+				if pr, found := prNums[pull.Number]; !found || pr.HeadRefOID != pull.SHA {
+					validPulls = false
+					break
+				}
+			}
+			if !validPulls {
+				continue
+			}
+			for _, pull := range pj.Spec.Refs.Pulls {
+				if _, ok := prowJobsForPRs[pull.Number]; !ok {
+					prowJobsForPRs[pull.Number] = prowJobsByContext{
+						successfulBatchJob:   make(map[string]prowapi.ProwJob),
+						pendingPresubmitJobs: make(map[string]bool),
+					}
+				}
+				prowJobsForPRs[pull.Number].successfulBatchJob[pj.Spec.Context] = pj
+			}
+		case prowapi.PresubmitJob:
+			if jobState != pendingState {
```

Comment:
do we also want to override failed presubmit runs if the batch job finishes with success?

Reply:
If a presubmit run of a batch member fails, the batch jobs continue to run. However, the batch is removed from the merge pool, so it would never reach this point here.